spec/with_locking_spec.rb

```diff
@@ -48,10 +49,40 @@
 
     it "checks if the file exists" do
       File.should_receive(:exists?).with(path)
+      File.should_receive(:exists?).with(path)
       WithLocking.locked?(file_name)
     end
 
   end
+
+  describe "#remove_staled_file" do
+
+    it "returns if the option :stale_seconds is not provided" do
+      File.should_receive(:exists?).with(path)
+      File.should_not_receive(:delete).with(path)
+      WithLocking.remove_stale_file(name: file_name)
+    end
+
+    it "deletes the file if stale_seconds is provided" do
+      sleep(2)
```

Comment:
I don't think sleeping is the right option. Instead, we should use something like [timecop](https://github.com/travisjeffery/timecop) to correctly work with time.